Source/TimeIntegration/StrainRate.cpp

```diff
@@ -136,3 +134,93 @@ ComputeStrainRate(const int &i, const int &j, const int &k,
 
   return strainRate;
 }
+
+AMREX_GPU_DEVICE
+Real
+ComputeExpansionRate(const int &i, const int &j, const int &k,
+                     const Array4<Real>& u, const Array4<Real>& v, const Array4<Real>& w,
+                     const enum NextOrPrev &nextOrPrev,
+                     const enum MomentumEqn &momentumEqn,
+                     const enum DiffusionDir &diffDir,
+                     const GpuArray<Real, AMREX_SPACEDIM>& cellSize) {
+    Real dx = cellSize[0];
+    Real dy = cellSize[1];
+    Real dz = cellSize[2];
+
+    Real expansionRate = 0;
+
+    //TODO: Check if it is more efficient to store computed divergence at cell-centers rather than
+    // computing on the fly
+
+    // Divergence at cell (i, j, k)
+    Real divergence = (u(i+1, j, k) - u(i, j, k))/dx +
```

Comment:
Move inside the switch so effort is not wasted computing this when it is not used. Even better, for readability, make a function (and force inlining for efficiency) to compute divergence, which could replace the other divergence computations below as well. 

Doing the above would also allow replacing all the `switch` conditionals with logic operations to compute the indices where divergence is computed (similar to here: https://github.com/erf-model/ERF/blob/724c3f50f46758c173326b52552a9c3a2aeb70d1/Source/TimeIntegration/Diffusion.cpp#L142), which I find to be more readable - but that's just my opinion, so it's not necessary.